src/org/zaproxy/zap/extension/httpsinfo/MDialog.java

```diff
@@ -0,0 +1,319 @@
+		}
+		if (server.getMaxStrength() == null) {
+			content.append(Constant.messages.getString("httpsinfo.general.achieveable.strength.heading") + " "
+					+ Constant.messages.getString("httpsinfo.general.pending.indicator"));
+		} else {
+			content.append(Constant.messages.getString("httpsinfo.general.achieveable.strength.heading") + " "
+					+ strengthString(server.getMaxStrength()) + NEWLINE);
+		}
+		if (server.getBeastVuln() == null) {
+			content.append(Constant.messages.getString("httpsinfo.general.beast.status.heading") + " "
+					+ Constant.messages.getString("httpsinfo.general.pending.indicator"));
+		} else {
+			content.append(
+					Constant.messages.getString("httpsinfo.general.beast.status.heading") + " "
+							+ (server.getBeastVuln()
+									? Constant.messages.getString("httpsinfo.general.vulnerable.indicator")
+									: Constant.messages.getString("httpsinfo.general.protected.indicator"))
+							+ NEWLINE);
+			if (server.getBeastVuln()) {
+				raiseBeastAlert();
+			}
+		}
+		if (server.getCrimeVuln() == null) {
+			content.append(Constant.messages.getString("httpsinfo.general.crime.status.heading")
+					+ Constant.messages.getString("httpsinfo.general.pending.indicator"));
+		} else {
+			content.append(
+					Constant.messages.getString("httpsinfo.general.crime.status.heading") + " "
+							+ (server.getCrimeVuln()
+									? Constant.messages.getString("httpsinfo.general.vulnerable.indicator")
+									: Constant.messages.getString("httpsinfo.general.protected.indicator")));
+			if (server.getCrimeVuln()) {
+				raiseCrimeAlert();
+			}
+		}
+
+		general.setText(content.toString());
+	}
+
+	public void showCipherSuites() {
+		StringBuffer cs = new StringBuffer(
+				Constant.messages.getString("httpsinfo.ciphersuites.supported.label") + " " + suppVers.getSelectedItem()
+						+ (ordered ? " " + Constant.messages.getString("httpsinfo.ciphersuites.ordered")
+								: " " + Constant.messages.getString("httpsinfo.ciphersuites.unordered")));
+		int x = 0;
+		try {
+			for (int c : server.getSupportedCipherSuites().get(versionStrings.get(suppVers.getSelectedItem()))) {
+				cs.append((ordered ? x + "." : "  ") + "      " + SSLServer.cipherSuiteString(c) + NEWLINE);
+				x++;
+			}
+
+		} catch (Exception e) {
+			cs.append("Pending...");
+			new Thread(new Runnable() {
+				@Override
+				public void run() {
+					server.updateSupportedCS();
+					server.updateAttackVuln();
+					server.updateStrength();
+					showGeneral();
+					showCipherSuites();
+				}
+			}).start();
+		}
+		cipherSuites.setText(cs.toString());
+	}
+
+	public void addComponentsToPane(Container pane) {
+		pane.setLayout(new GridBagLayout());
+		pane.setPreferredSize(new Dimension(750, 400));
+		pane.setMinimumSize(new Dimension(750, 400));
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.BOTH;
+		c.gridwidth = 3;
+		c.anchor = GridBagConstraints.NORTH;
+		c.gridx = 0;
+		c.gridy = 0;
+		pane.add(general, c);
```

Comment:
Yeah, there's no definition how the extra vertical space is distributed for the components. Setting `c.weighty = 0.5;` for both the text areas (or maybe 0.3/0.7 split since the lower text area shows more info?) and none for the buttons/labels `c.weighty = 0;`, helps with that.
The JScrollPane can be added as usual:
`pane.add(new JScrollPane(general), c);`

Also, the statements:
```Java
pane.setPreferredSize(new Dimension(750, 400));
pane.setMinimumSize(new Dimension(750, 400));
```
could be removed and the preferred size controlled with the text areas, for example:
```Java
general.setRows(10);
general.setColumns(45);
```